File: src/main/java/net/snipsniper/secrets/games/BGameResources.java
package net.snipsniper.secrets.games;

import net.snipsniper.ImageManager;

import java.awt.*;
import java.util.ArrayList;

public class BGameResources {
    private final ArrayList<int[][]> pieces = new ArrayList<>();
    private final ArrayList<Image> sprites = new ArrayList<>();

    public void init() {
        pieces.add(new int[][]{
                {0,0,1,0},
                {0,0,1,0},
                {0,0,1,0},
                {0,0,1,0}
        });
        sprites.add(ImageManager.Companion.getImage("icons/snipsniper.png"));

        pieces.add(new int[][]{
                {1,1},
                {1,1}
        });
        sprites.add(ImageManager.Companion.getImage("icons/editor.png"));

        pieces.add(new int[][]{
                {1,0,0},
                {1,0,0},
                {1,1,0}
        });
        sprites.add(ImageManager.Companion.getImage("icons/viewer.png"));

        pieces.add(new int[][]{
                {1,0,0},
                {1,1,0},
                {1,0,0}
        });
        sprites.add(ImageManager.Companion.getImage("icons/random/kiwi.png"));

        pieces.add(new int[][]{
                {0,1,0},
                {0,1,0},
                {1,1,0}
        });
        sprites.add(ImageManager.Companion.getImage("icons/console.png"));

        pieces.add(new int[][]{
                {0,1,0},
                {1,1,0},
                {1,0,0}
        });
        sprites.add(ImageManager.Companion.getImage("icons/about.png"));

        pieces.add(new int[][]{
                {1,0,0},
                {1,1,0},
                {0,1,0}
        });
        sprites.add(ImageManager.Companion.getImage("icons/folder.png"));
    }

    public Image getImage(int index) {
        return sprites.get(index);
    }

    public int[][] getPiece(int index) {
        return pieces.get(index);
    }

    public int getSize() {
        return pieces.size();
    }
}

File: src/main/kotlin/net/snipsniper/secrets/games/BGameResources.kt
package net.snipsniper.secrets.games

import net.snipsniper.utils.getImage
import java.awt.Image

class BGameResources {
    private val pieces = ArrayList<Array<IntArray>>()
    private val sprites = ArrayList<Image>()
    fun init() {
        pieces.add(arrayOf(
                intArrayOf(0, 0, 1, 0),
                intArrayOf(0, 0, 1, 0),
                intArrayOf(0, 0, 1, 0),
                intArrayOf(0, 0, 1, 0)
        ))
        sprites.add("icons/snipsniper.png".getImage())

        pieces.add(arrayOf(
            intArrayOf(1, 1),
            intArrayOf(1, 1)
        ))
        sprites.add("icons/editor.png".getImage())

        pieces.add(arrayOf(
            intArrayOf(1, 0, 0),
            intArrayOf(1, 0, 0),
            intArrayOf(1, 1, 0)
        ))
        sprites.add("icons/viewer.png".getImage())

        pieces.add(arrayOf(
            intArrayOf(1, 0, 0),
            intArrayOf(1, 1, 0),
            intArrayOf(1, 0, 0)
        ))
        sprites.add("icons/random/kiwi.png".getImage())

        pieces.add(arrayOf(
            intArrayOf(0, 1, 0),
            intArrayOf(0, 1, 0),
            intArrayOf(1, 1, 0)
        ))
        sprites.add("icons/console.png".getImage())

        pieces.add(arrayOf(
            intArrayOf(0, 1, 0),
            intArrayOf(1, 1, 0),
            intArrayOf(1, 0, 0)
        ))
        sprites.add("icons/about.png".getImage())

        pieces.add(arrayOf(
            intArrayOf(1, 0, 0),
            intArrayOf(1, 1, 0),
            intArrayOf(0, 1, 0)
        ))
        sprites.add("icons/folder.png".getImage())
    }

    fun getImage(index: Int) = sprites[index]

    fun getPiece(index: Int) = pieces[index]

    fun getSize() = pieces.size
}